### PR TITLE
Snap 1529 -- Job server changes to accomodate dependent URIs

### PR DESCRIPTION
**Pull request by kneeraj**
Changes to honour passed dependent URI for class loading needed by submitted jobs.

### CODE REVIEW

File: job-server/src/org/apache/spark/deploy/PackageAndDepUtils.scala [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    println(PackageAndDepUtils.resolveMavenCoordinates(coordinates, remoteRepos, ivyPath))
+  }
+}
+
+object PackageAndDepUtils {
+  def resolveMavenCoordinates(coordinates: String, remoteRepos: Option[String],
+        ivyPath: Option[String], exclusions: Seq[String] = Nil, isTest: Boolean = false): String = {
+    SparkSubmitUtils.resolveMavenCoordinates(coordinates, remoteRepos, ivyPath, exclusions, isTest)
+  }
+}
```
Comment (rishitesh) [inline job-server/src/org/apache/spark/deploy/PackageAndDepUtils.scala]: A new line as per convention.

File: job-server/src/org/apache/spark/deploy/PackageAndDepUtils.scala [hunk 2]
```diff
@@ -0,0 +1,44 @@
+package org.apache.spark.deploy
+
+object GetJarsAndDependencies {
```
Comment (rishitesh) [inline job-server/src/org/apache/spark/deploy/PackageAndDepUtils.scala]: As discussed this class can be moved to cluster module.

File: job-server/src/spark.jobserver/JobManagerActor.scala [hunk 3]
```diff
@@ -150,12 +150,18 @@ class JobManagerActor(contextConfig: Config) extends InstrumentedActor {
...
-        val jarToLoad = new URL(convertJarUriSparkToJava(jarUri))
-        if(! loadedJars.contains(jarToLoad)){
-          logger.info("Adding {} to Current Job Class path", jarUri)
-          wrappedLoader.addURL(new URL(convertJarUriSparkToJava(jarUri)))
-          // jobContext.sparkContext.addJar(jarUri)
+      logger.info(s"dependent-jar-uris" +
+          s" ${Try(jobConfig.getString
+          ("dependent-jar-uris").split('|').toSeq).getOrElse(Seq.empty[String])}")
+      val allDepJars = Try(jobConfig.getString
+      ("dependent-jar-uris").split('|').toSeq).getOrElse(Seq.empty[String])
```
Comment (rishitesh) [inline job-server/src/spark.jobserver/JobManagerActor.scala:156]: Why did we move from the comma separated to "|" separated jars?

**Review by rishitesh — APPROVED** [review]
One clarification sought.

File: job-server/src/spark.jobserver/JobManagerActor.scala (hunk 3 above)
Comment (kneeraj) [reply job-server/src/spark.jobserver/JobManagerActor.scala:156]: Don't remember the exact error I was getting when I used ','. The splitter on the job-server side confused and was not splitting it properly. Changing it to '|' worked.

**Pull request merged**